### PR TITLE
#748 Iterables for primitive types

### PR DESCRIPTION
**Pull request by Vatavuk**
Introduced iterables for primitive types for #748

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#786](https://github.com/yegor256/cactoos/issues/786) is assigned to @sharpler[/z](https://www.0crat.com/u/sharpler), here is [why](https://www.0crat.com/footprint/C63314D6Z/1141866717); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by SharplEr** [comment]
@VsSekorin I think the design of your solution is not optimal. Please see how `IteratorOf` class works. I think you can avoid copying into `ArrayList` by creating a new instance of `Iterator` with counter and primitive array inside.

**Comment by llorllale** [comment]
@Vatavuk tip: remember the budget is only 30 mins

**Comment by Vatavuk** [comment]
@llorllale I've coded with timer. It took me 32min:D. I will definitely create more puzzles for `IteratorsOf` if it would be the case.

**Comment by Vatavuk** [comment]
@SharplEr I've created 2 puzzles for `Iterator*` classes. I think that is better to expose them to the user (to give him more design freedom) than to encapsulate them in `Iterable*` classes.

File: src/main/java/org/cactoos/iterable/IterableOfBooleans.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+public final class IterableOfBooleans extends IterableEnvelope<Boolean> {
+
+    /**
+     * Ctor.
+     * @param values Boolean values
+     */
+    public IterableOfBooleans(final boolean... values) {
+        super(() -> {
+            final Collection<Boolean> iterable =
+                new ArrayList<>(values.length - 1);
```
Comment (SharplEr) [inline src/main/java/org/cactoos/iterable/IterableOfBooleans.java]: @Vatavuk Why do you decrement here? The code of this constructor is:
```Java
    public ArrayList(int initialCapacity) {
        if (initialCapacity > 0) {
            this.elementData = new Object[initialCapacity];
        } else if (initialCapacity == 0) {
            this.elementData = EMPTY_ELEMENTDATA;
        } else {
            throw new IllegalArgumentException("Illegal Capacity: "+
                                               initialCapacity);
        }
    }
```
So if you need `n` elements in `ArrayList` just call `new ArrayList<>(n)`.

**Comment by SharplEr** [comment]
@Vatavuk  Okay then. But I think the description of puzzles a bit unclear. Please add something like `avoid to array copy` to help to explain which problem exists in the current implementation. Also, I asked one thing in code. After that, we can merge:)

**Comment by Vatavuk** [comment]
@SharplEr fixed in 5dd929c

File: src/main/java/org/cactoos/iterable/IterableOfBooleans.java (hunk 1 above)
Comment (Vatavuk) [reply src/main/java/org/cactoos/iterable/IterableOfBooleans.java]: @SharplEr fixed in 5dd929c

**Comment by SharplEr** [comment]
@llorllale good to merge

**Comment by llorllale** [comment]
@Vatavuk there is a merge conflict now

**Comment by Vatavuk** [comment]
@llorllale merge conflicts fixed

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@d20feeb0-0ee6-4a68-b678-bb443d7a88c9, thread: AsyncFlush-2
com.zerocracy.farm.reactive.StkGroovy[125] java.lang.IllegalStateException: java.lang.reflect.InvocationTargetException in com/zerocracy/stk/pm/in/orders/close_job.groovy
sun.reflect.NativeMethodAccessorImpl[-2] java.lang.reflect.InvocationTargetException: null
com.amazonaws.http.AmazonHttpClient$RequestExecutor[1113] com.amazonaws.SdkClientException: Unable to execute HTTP request: null
sun.nio.ch.FileChannelImpl[110] java.nio.channels.ClosedChannelException: null
```

[1.0-SNAPSHOT](https://github.com/zerocracy/farm)

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @sharpler[/z](https://www.0crat.com/u/sharpler), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#786](https://github.com/yegor256/cactoos/issues/786) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@quality good

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @sharpler[/z](https://www.0crat.com/u/sharpler)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**